app/controllers/split_checkout_controller.rb

```diff
@@ -21,6 +21,10 @@ class SplitCheckoutController < ::BaseController
 
   def edit
     redirect_to_step_based_on_order unless params[:step]
+
+    return if available_shipping_methods.any?
+
+    flash[:error] = I18n.t('split_checkout.errors.no_shipping_methods_available')
```

Comment:
This is a bit confusing because it looks like the error case is the normal flow and the success case is terminating early. Would this be better?
```suggestion
    if available_shipping_methods.none?
      flash[:error] = I18n.t('split_checkout.errors.no_shipping_methods_available')
      return
    end
```

I could imagine rubocop complaining about it. We could also move the action to a method. I'm not sure without trying it out.

Reply:
Yes, rubocop complains, twice: 
 - Use a guard clause (`return unless available_shipping_methods.none?`) instead of wrapping the code inside a conditional expression. (convention:Style/GuardClause)
 - Redundant `return` detected

I'm wondering if this will do to the trick :
```ruby
    return unless available_shipping_methods.none?
    
    flash[:error] = I18n.t('split_checkout.errors.no_shipping_methods_available')
```

I personally dislike the `return unless` statement, which I find confusing, I do prefer use the `return if` statement.

So, what about
```ruby
flash_error_when_no_shipping_method_available if available_shipping_methods.none?

...

  def flash_error_when_no_shipping_method_available
    flash[:error] = I18n.t('split_checkout.errors.no_shipping_methods_available')
  end
```

See commit f0f287259

Reply:
Yes. that's a good solution.